cmd/root.go

```diff
@@ -1,24 +1,40 @@
 package cmd
 
 import (
+	"context"
+	"fmt"
 	"os"
 	"path/filepath"
+	"strings"
+	"time"
 
+	crdaConfig "github.com/fabric8-analytics/cli-tools/pkg/config"
+	"github.com/fabric8-analytics/cli-tools/pkg/segment"
+	"github.com/fabric8-analytics/cli-tools/pkg/telemetry"
 	"github.com/fatih/color"
-	homedir "github.com/mitchellh/go-homedir"
+	"github.com/mitchellh/go-homedir"
 	"github.com/rs/zerolog"
 	"github.com/rs/zerolog/log"
 
-	constants "github.com/fabric8-analytics/cli-tools/utils"
+	"github.com/fabric8-analytics/cli-tools/utils"
 	"github.com/spf13/cobra"
 	"github.com/spf13/viper"
 )
 
 // Flags
 var (
 	debug       bool
-	cfgFile     string
 	flagNoColor bool
+	jenkins     bool
+	tekton      bool
+	ghActions   bool
```

Comment:
Why can't this be a string instead of bool switch per integration? This would make a new integration requiring a change in cli.

Reply:
In that case, any typo, like jenkings, github-actions, will flow in telemetry and mess up the data

Reply:
Again, it is a tradeoff between flexibility vs correctness. I feel the correctness shall be established here because of the controlled environment.

Reply:
I would like to ask @prashbnair take on this. We should be aware of tradeoffs in adopting each case

Reply:
@prashbnair as an env variable or as a param?

Reply:
@yzainee It would be a Param any way, to be more specific it will be a flag .. after discussion:
We are going ahead with `crda analyse --mode=jenkins`

You can always override this by providing `mode` value in Jenkins Env.

Reply:
As this value is static, im fine with either env or flag.

Reply:
> I think a mode flag can be added with values like jenkins, tekton and gh-actions

Ack. `client` flag added along with value validators